gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -43,7 +47,7 @@
  */
 public class SftpFsHelper implements SizeAwareFileBasedHelper {
   private static Logger log = LoggerFactory.getLogger(SftpFsHelper.class);
-  private ChannelSftp channelSftp;
+  Set<Channel> channels = Sets.newConcurrentHashSet();
```

Comment:
Channel's API does not provide a way to say if it is being used or not. 
Channel#isClosed(), Channel#isConnected() are the only two API's provided. 
Now if we want to reuse a channel we have to call Channel#disconnect() after every action so the next action can re-connect and use it. Since connection and disconnection is the 'heavier' operation I do not see value in reusing. 
If garbage collection the concern? If so I can use pool.